commons/src/main/java/org/infinispan/commons/util/IntSet.java

```diff
@@ -0,0 +1,270 @@
+      }
+      return false;
+   }
+
+   /**
+    * Add an integer to the set without boxing the parameter or checking if the integer was already present in the set.
+    */
+   public void set(int i) {
+      bitSet.set(i);
+   }
+
+   /**
+    * If {@code value} is {@code true}, add the integer to the set, otherwise remove the integer from the set.
+    */
+   public void set(int i, boolean value) {
+      bitSet.set(i, value);
+   }
+
+   @Override
+   public boolean remove(Object o) {
+      if (!(o instanceof Integer)) {
+         return false;
+      }
+      return remove((int) o);
+   }
+
+   /**
+    * Remove an integer from the set without boxing.
+    */
+   public boolean remove(int i) {
+      boolean wasSet = bitSet.get(i);
+      if (wasSet) {
+         bitSet.clear(i);
+         return true;
+      }
+      return false;
+   }
+
+   @Override
+   public boolean containsAll(Collection<?> c) {
+      return c.stream().allMatch(e -> contains(e));
+   }
+
+   @Override
+   public boolean addAll(Collection<? extends Integer> c) {
+      boolean modified = false;
+      for (Integer integer : c) {
+         modified |= add(integer);
+      }
+      return modified;
+   }
+
+   @Override
+   public boolean removeAll(Collection<?> c) {
+      boolean modified = false;
+      for (Object integer : c) {
+         modified |= remove(integer);
+      }
+      return modified;
+   }
+
+   @Override
+   public boolean retainAll(Collection<?> c) {
+      boolean modified = false;
+      for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
+         if (!c.contains(i)) {
+            bitSet.clear(i);
+            modified = true;
+         }
+      }
+      return modified;
+   }
+
+   @Override
+   public void clear() {
+      bitSet.clear();
+   }
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o)
+         return true;
+      if (o == null || !(o instanceof Set))
+         return false;
+
+      if (o instanceof IntSet) {
+         IntSet integers = (IntSet) o;
+         return bitSet.equals(integers.bitSet);
+      } else {
+         Set set = (Set) o;
+         return size() == set.size() && containsAll(set);
+      }
+   }
+
+   @Override
+   public int hashCode() {
+      return bitSet.hashCode();
+   }
+
+   @Override
+   public String toString() {
+      StringBuilder sb = new StringBuilder("{");
+      for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
+         if (sb.length() > "{".length()) {
+            sb.append(' ');
+         }
+         int runStart = i;
+         while (bitSet.get(i + 1)) {
+            i++;
+         }
+         if (i == runStart) {
+            sb.append(i);
+         } else {
+            sb.append(runStart).append('-').append(i);
+         }
+      }
+      sb.append('}');
+      return sb.toString();
+   }
+
+   public static void writeTo(ObjectOutput output, IntSet set) throws IOException {
+      output.writeInt(set.size());
```

Comment:
Sorry, I was thinking `UnsignedNumeric` is about handling the sign properly for values > `Integer.MAX_VALUE`. I take it you just meant I could be using fewer bytes, @pruivo?